/*
    @Author: Bhaskar S
    @Blog:   https://www.polarsparc.com
    @Date:   04 Jul 2020
 */

syntax = "proto3";

package com.polarsparc.protobuf3.complex;

option java_outer_classname = "CustomerAccount";

message Customer {
  string first_name = 1;
  string last_name = 2;
  string email_id = 3;

  int32 age = 4;

  repeated string phone_no = 5;
}

enum AccountType {
  CA_UNKNOWN = 0;
  CA_SAVINGS = 1;
  CA_CHECKING = 2;
  CA_BROKERAGE = 3;
}

message Account {
  string acct_no = 1;

  AccountType acct_type = 2;

  Customer customer = 3;
}
